File: cmms php/scripts/read_xlsx.py
import zipfile
import xml.etree.ElementTree as ET

filename = r'c:\Users\star_\OneDrive\Escritorio\Prueba 2.xlsx'

try:
    with zipfile.ZipFile(filename, 'r') as z:
        shared_strings = []
        if 'xl/sharedStrings.xml' in z.namelist():
            with z.open('xl/sharedStrings.xml') as f:
                tree = ET.parse(f)
                root = tree.getroot()
                # The namespace is usually required
                ns = {'x': 'http://schemas.openxmlformats.org/spreadsheetml/2006/main'}
                for si in root.findall('x:si', ns):
                    t = si.find('x:t', ns)
                    val = t.text if t is not None else ""
                    shared_strings.append(val)

        if 'xl/worksheets/sheet1.xml' in z.namelist():
            with z.open('xl/worksheets/sheet1.xml') as f:
                tree = ET.parse(f)
                root = tree.getroot()
                ns = {'x': 'http://schemas.openxmlformats.org/spreadsheetml/2006/main'}
                sheetData = root.find('x:sheetData', ns)
                if sheetData is not None:
                    first_row = sheetData.find('x:row', ns)
                    if first_row is not None:
                        headers = []
                        for c in first_row.findall('x:c', ns):
                            v = c.find('x:v', ns)
                            val = v.text if v is not None else ""
                            if c.get('t') == 's' and val.isdigit():
                                val = shared_strings[int(val)]
                            headers.append(val)
                        print("HEADERS:")
                        for i, h in enumerate(headers):
                            print(f"{i+1}: {h}")
except Exception as e:
    print("Error:", e)

File: cmms php/Backend/Archive/Legacy_Maintenance/read_xlsx.php
<?php
$file = 'c:\\Users\\star_\\OneDrive\\Escritorio\\Prueba 2.xlsx';
$zip = new ZipArchive;
if($zip->open($file) === true){
    echo "Archivo XLSX abierto correctamente.\n";
    $ss=[];
    if($x=$zip->getFromName('xl/sharedStrings.xml')){
        $xml = simplexml_load_string($x);
        foreach($xml->si as $s) {
            $ss[]=(string)$s->t;
        }
    }
    if($y=$zip->getFromName('xl/worksheets/sheet1.xml')){
        $xml = simplexml_load_string($y);
        $firstRow = $xml->sheetData->row[0];
        $h=[];
        foreach($firstRow->c as $c){
            $v=(string)$c->v;
            if((string)$c['t']=='s') $v=$ss[(int)$v];
            $h[]=$v;
        }
        echo "CABECERAS EN XLSX:\n";
        print_r($h);
    } else {
        echo "No se encontro sheet1 en el XLSX\n";
    }
    $zip->close();
} else {
    echo "Falló la apertura del archivo XLSX\n";
}
?>
